Documentation/class-libraries.md

```diff
@@ -0,0 +1,50 @@
+
+There are three types of class libraries that you can use:
+
+- **Platform-specific** class libraries have access to all the APIs in a given platform (e.g. .NET Framework, Xamarin iOS), but can only be used by apps and libraries that target that platform.
+- **Portable** class libraries have access to a subset of APIs, and can be used by apps and libraries that target multiple platforms.
+- **.NET Core** class libraries are a merger of the platform-specific and portable library concept into a single model that provides the best of both.
+
+Platform-specific Class Libraries
+=================================
+
+Platform-specific libraries are bound to a single .NET platform (e.g. .NET Framework on Windows) and can therefore take significant dependencies on a known execution environment. Such an environment will expose a known set of APIs (.NET and OS APIs) and will maintain and expose expected state (e.g. Windows registry). 
+
+Developers who create plaform specific libraries can fully exploit the underlying platform. The libraries will only ever run on that given platform, making platform checks or other forms of conditional code unnecessary (modulo single sourcing code for multiple platforms).
+
+Platform-specific libraries have been the primary class library type for the .NET Framework. Even as other .NET platforms emerged, platform-specific libraries remained the dominant library type.
+
+Portable Class Libraries
+========================
+
+Portable libraries are supported on multiple .NET platforms. They can still take dependencies on a known execution environment, however, the environment is a synthetic one that is generated by the intersection of a set of concrete .NET platforms. This means that exposed APIs and platform assumptions are a subset of what would be available to a platform-specific library. 
+
+You choose a platform configuration when you create a portable library. These are the set of platforms that you need to support (e.g. .NET Framework 4.5+, Windows Phone 8.0+). The more platforms you opt to support, the fewer APIs and fewer platform assumptions you can make, the lowest common denominator. This characteristic can be confusing at first, since people often think "more is better", but find that more supported platforms results in fewer available APIs.
+
+Many library developers have switched from producing multiple platform-specific libraries from one source (using conditional compilation directives) to portable libraries. There are several techniques for accessing platform-specific functionality within portable libraries.
```

Comment:
I'd recommend listing or linking to descriptions of the techniques to access platform specific functionality from a PCL.  A good overview of the techniques is [here](http://blog.stephencleary.com/2012/11/portable-class-library-enlightenment.html).  Over the past year, the [bait and switch](http://log.paulbetts.org/the-bait-and-switch-pcl-trick/) approach has gained popularity and seems to be the most widely accepted technique for reusable libraries.